main.py

```diff
@@ -0,0 +1,244 @@
+        with open(task_metadata_path, 'rb') as pickle_file:
+            task_metadata = pickle.load(pickle_file)
+
+        dfs = []
+
+        page = 100
+        while True:
+            logging.debug('Reading Neurovault page #{:d}'.format(page))
+            page_url = NEUROVAULT_URL + '?offset=%d' % page
+
+            with urllib.request.urlopen(page_url) as url:
+                data = json.loads(url.read().decode())
+                page_results = pd.io.json.json_normalize(data['results'])
+                dfs.append(page_results)
+                if data['next'] is 'null':
+                    print('Found null')
+                    break
+                page += 100
+
+                if DEBUG:
+                    if page > 35000:
+                        break
+
+            img_metadata = pd.concat(dfs, ignore_index=True)
+            logging.info('{:d} images metadata returned by the query'.format(
+                len(img_metadata)))
+            logging.debug('Image metadata fields are: \n{}'.format(
+                img_metadata.keys()))
+
+        task_ids = task_metadata.id
+        unique_task_ids = set(task_ids)
+        n_task_ids = len(unique_task_ids)
+        assert n_task_ids == len(task_ids)
+        logging.info('Cognitive atlas has {:d} unique mental task IDs'.format(
+            n_task_ids))
+
+        labels = img_metadata.cognitive_paradigm_cogatlas_id
+        n_imgs = len(labels)
+        # Bring the NaN to None
+        labels = labels.where((pd.notnull(labels)), None)
+
+        mask_not_none = ~np.equal(labels, None)
+        not_none_labels = labels[mask_not_none]
+        n_imgs_with_id = len(not_none_labels)
+        unique_labels = set(not_none_labels)
+
+        logging.info('{:d}/{:d} images have paradigm ID, '
+                     'taken among {:d} unique IDs.'.format(
+                         n_imgs_with_id, n_imgs, len(unique_labels)))
+
+        n_intersection = len(unique_task_ids & unique_labels)
+
+        mask_in_task_id = np.isin(
+            np.array(labels), list(unique_task_ids))
+        logging.info('{:d}/{:d} images have paradigm ID corresponding'
+                     'to a mental task ID in the Cognitive Atlas, '
+                     ' which corresponds to {:d} unique IDs available '
+                     'in Cognitive Atlas'.format(
+                         sum(mask_in_task_id), n_imgs, n_intersection))
+
+        img_files = img_metadata['file'].loc[
+            mask_in_task_id].values
+        img_files = np.expand_dims(img_files, axis=1)
+        img_ids = img_metadata['id'].loc[
+            mask_in_task_id].values
+        img_ids = np.expand_dims(img_ids, axis=1)
+        img_task_ids = img_metadata['cognitive_paradigm_cogatlas_id'].loc[
+            mask_in_task_id].values
+        img_task_ids = np.expand_dims(img_task_ids, axis=1)
+
+        assert len(img_files) == len(img_task_ids)
+
+        csv_rows = np.hstack([img_files, img_ids, img_task_ids])
```

Comment:
since you're already using pandas - does it make sense to use it all the way here and do the writing with pandas? I don't remember if we already discussed that...

Reply:
Yes, thanks.